flexvolume/flexvolume.go

```diff
@@ -67,59 +67,71 @@ func finish(Response Response) {
 
 func RunPlugin(plugin FlexVolumePlugin) {
 	if len(os.Args) < 2 {
-		finish(Fail("Expected at least one argument"))
+		finish(Fail("expected at least one argument"))
 	}
 
 	switch os.Args[1] {
 	case "init":
 		finish(plugin.Init())
 
 	case "attach":
-		if len(os.Args) != 3 {
-			finish(Fail("attach expected exactly 3 arguments; got ", os.Args))
-		}
+		finish(attach(plugin, os.Args))
 
-		opt := plugin.NewOptions()
-		if err := json.Unmarshal([]byte(os.Args[2]), opt); err != nil {
-			finish(Fail("Could not parse options for attach; got ", os.Args[2]))
-		}
+	case "detach":
+		finish(detach(plugin, os.Args))
 
-		finish(plugin.Attach(opt))
+	case "mount":
+		finish(mount(plugin, os.Args))
 
-	case "detach":
-		if len(os.Args) != 3 {
-			finish(Fail("detach expected exactly 3 arguments; got ", os.Args))
-		}
+	case "unmount":
+		finish(unmount(plugin, os.Args))
 
-		device := os.Args[2]
-		finish(plugin.Detach(device))
+	default:
+		finish(Fail(fmt.Sprintf("not sure what to do. Called with: %v", os.Args)))
+	}
+}
 
-	case "mount":
-		if len(os.Args) != 5 {
-			finish(Fail("mount expected exactly 5 argument; got ", os.Args))
-		}
+func attach(plugin FlexVolumePlugin, args []string) Response {
+	if len(args) != 3 {
+		finish(Fail(fmt.Sprintf("attach expected exactly 3 arguments; got %v", args)))
```

Comment:
You can't `finish()` here is you finish already when calling the function.
`attach()` should return a `Response`